cmd/kube-controller-manager/app/controllermanager.go

```diff
@@ -414,6 +414,13 @@ var ControllersDisabledByDefault = sets.NewString(
 	"tokencleaner",
 )
 
+// GenericControllers is the set of controllers which watch all resources. Generic controllers are started
+// after others to reuse typed informers initialized by others.
+var GenericControllers = sets.NewString(
+	"resourcequota",
```

Comment:
I think we can use WaitGroup here.

For controllers except for GC and ResourceQuota:
```
wg.Add(len(controllersToStart))

# for each controller
StartController()
wg.Done()
```

For GC and ResourceQuota:

```
# pass wg into controller initialization func
wg.Add(1)
startResourceQuotaController(wg)

# inside startResourceQuotaController()
wg.Wait()
startDiscoveryAndCreateMetadataInformers()
```

But to use WG, we have to make initialization functions of GC and ResourceQuota non-blocking. WDYT?

Reply:
Now that the code has changed a bit and that makes the starting order easier to manipulate. However,  it's still not like what you suggested because of the complexity. Anyway, please have a look.